### ISSUE TITLE
possibility to exclude a set of system:partition on generic runs

### ISSUE THREAD

**Issue by akesandgren** [reframe-hpc/reframe#2758]
It would sometimes be useful to be able to exclude specific system:partition's from a generic run.
For instance running a test which has "valid_systems = ['+gpu']" but excluding specific gpu enabled partitions  with --exclude-system "*:V100" or some other regex.

**Comment by vkarak** [comment]
Couldn't that partition have a unique feature defined, so that you could then say `-feat` in `valid_systems` to exclude it? Then you could also set the `valid_systems` for this test explicitly from the command line as well.

**Comment by akesandgren** [comment]
Nah, the situation I'm thinking of is that I have multiple partitions which almost the same, features are "gpu" for instance, and I want to run the test on all gpu partitions except one (or a few) with a specific name. Mainly because I know there are no available nodes in those at that specific time. So I have 8 partitions to run stuff on but one have no free nodes... and I don't want to have to list the 7 I want on the cmdline
